UMN-AutoPackager/public/Set-UMNGlobalConfig.ps1

```diff
@@ -0,0 +1,153 @@
+                        }
+                        catch {
+                           Write-Warning -Message "Failed to update JSON file. $($_Exception.message)"
+                        }
+                    }
+                }
+                if ($config.$k.gettype().Name -eq "Object[]") {
+                   Write-Verbose -Message "$k is Array"
+                   if ($PSCmdlet.ParameterSetName -eq "Main") {
+                       Write-Verbose -Message "Parameter set is Main"
+                       # add a parameter check for Configmgr
+                        if ($k -eq $Key) {
+                            Write-Verbose -Message "Found match for $k"
+                            if ($config.$k -contains $value) {
+                                # Move these tests to the specific if statements
+                                Write-Verbose -Message "Already contains $value"
+                            }
+                            elseif ($k -eq "RecipeLocations") {
+                                # Maybe make a new Parameter set for recipelocations and add locationType
+                                $config.$k += New-Object -TypeName PSObject -Property ([ordered]@{locationType="directory";locationUri=$Value})
+                                $config.LastModified = (Get-Date).ToString()
+                                try {
+                                    $config | convertto-json | Set-Content -Path $path
+                                Write-Verbose -Message "$key is updated with $value"
+                                }
+                                catch {
+                                Write-Warning -Message "Failed to update JSON file. $($_Exception.message)"
+                                }
+                            }
+                            else {
+                                [System.collections.ArrayList]$new = $config.$k
+                                $new.add("$value") > $null
+                                $config.$k = $new
+                                $config.LastModified = (Get-Date).ToString()
+                                try {
+                                    $config | convertto-json | Set-Content -Path $path
+                                Write-Verbose -Message "$key is updated with $value"
+                                }
+                                catch {
+                                Write-Warning -Message "Failed to update JSON file. $($_Exception.message)"
+                                }
+                            }
+                        }
+                   }
+                }
+            } #foreach $key
+        }
+        else {
+            Write-Warning -Message "$Key is not a valid Key."
+        }
+    }
+    end {
+        Write-Verbose -Message "Ending $($myinvocation.mycommand)"
+    }
+}#Set-Config
+Set-UMNGlobalConfig -Path C:\Users\thoen008\Desktop\GlobalConfig.json -Key "RecipeLocations" -Value "C:\temp2" -Verbose
```

Comment:
I think this last line was from your testing and you might not have intended to include it.

Reply:
Yeah. I was committing changes while still working out the functionality. So you probably saw one of those iterations. I have removed them in the last commit.

Reply:
That's what I figured. Resolving.